pkg/config/config.go

```diff
@@ -388,3 +422,106 @@ func checkAdvertiseAddressConfigured(advertiseAddress string) error {
 	}
 	return fmt.Errorf("Advertise address: %s not present in any interface", advertiseAddress)
 }
+
+func validateRouterListenAddress(entries []string, advertiseAddress string, skipInterface bool) error {
+	addresses, err := AllowedListeningIPAddresses()
+	if err != nil {
+		return err
+	}
+	names, err := AllowedNICNames()
+	if err != nil {
+		return err
+	}
+	for _, entry := range entries {
+		if net.ParseIP(entry) != nil {
+			if entry == advertiseAddress && !skipInterface {
+				continue
+			}
+			if !slices.Contains(addresses, entry) {
+				return fmt.Errorf("IP %v not present in the host", entry)
+			}
+		} else if slices.Contains(names, entry) {
+			continue
+		} else {
+			return fmt.Errorf("entry %v not present in the host", entry)
+		}
+	}
+	return nil
+}
+
+func getBannedIPs() ([]*net.IPNet, error) {
+	banned := make([]*net.IPNet, 0)
+	for _, entry := range defaultRouterForbiddenCIDRs {
+		_, netIP, err := net.ParseCIDR(entry)
+		if err != nil {
+			return nil, err
+		}
+		banned = append(banned, netIP)
+	}
+	return banned, nil
+}
+func getHostAddresses() ([]net.IP, error) {
+	handle, err := netlink.NewHandle()
+	if err != nil {
+		return nil, err
+	}
+	links, err := handle.LinkList()
+	if err != nil {
+		return nil, err
+	}
+	addresses := make([]net.IP, 0)
+	for _, link := range links {
+		if link.Attrs().ParentIndex != 0 || link.Attrs().MasterIndex != 0 {
+			continue
+		}
+		addressList, err := handle.AddrList(link, netlink.FAMILY_V4)
+		if err != nil {
+			return nil, err
+		}
+		for _, addr := range addressList {
+			addresses = append(addresses, addr.IP)
+		}
+	}
+	return addresses, nil
+}
+
+func AllowedListeningIPAddresses() ([]string, error) {
+	bannedAddresses, err := getBannedIPs()
+	if err != nil {
+		return nil, err
+	}
+	hostAddresses, err := getHostAddresses()
+	if err != nil {
+		return nil, err
+	}
+	addressList := make([]string, 0)
```

Comment:
At worse it's slightly too big, but single allocation instead of several allocs
```suggestion
	addressList := make([]string, 0, len(hostAddresses))
```

Reply:
Same comment as above